kube-runtime/src/watcher.rs

```diff
@@ -75,6 +75,36 @@ impl<K> Event<K> {
         }
         .into_iter()
     }
+
+    /// Map each object in an event through a mutable fn
+    ///
+    /// This allows for memory optimizations in watch streams.
+    /// If you are chaining a watch stream into a reflector as an in memory state store,
+    /// you can control the space used by each object by dropping fields.
+    ///
+    /// ```no_run
+    /// use k8s_openapi::api::core::v1::Pod;
+    /// use kube::ResourceExt;
+    /// # use kube::runtime::watcher::Event;
+    /// # let event: Event<Pod> = todo!();
+    /// event.map_event(|pod| {
+    ///     pod.managed_fields_mut().clear();
+    ///     pod.annotations_mut().clear();
+    ///     pod.status = None;
+    /// });
+    /// ```
+    pub fn map_event(mut self, f: impl Fn(&mut K)) -> Self {
```

Comment:
`mutate_objects`, perhaps? `map` implies that it operates on a `fn(T) -> T` rather than `fn(&mut T) -> ()`

Reply:
I believe this could also be `FnMut` rather than `Fn` since we apply them sequentially (unless you plan on introducing parallelism later?)

Reply:
AFAIKT the benefit to having it be `FnMut` is that it can capture mutable variables from a closure?

Reply:
Yes. Or rather, it can capture closure variables mutably. You could still capture mutable variables immutably since they just get downgraded to immutable during the borrow.